dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java

```diff
@@ -363,4 +387,62 @@ public void testDeleteVersion_KeepPermissions() throws DotSecurityException, Dot
         }
     }
 
+    @Test
+    @UseDataProvider("testCases")
+    public void testIndexSearchFilteringByDates(final TestCase testCase){
+        final long languageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+
+        LocalDateTime today, yesterday, tomorrow;
+
+        final LocalDateTime now = LocalDateTime.now();
+
+        if (testCase.formatPattern.equals("hh:mm:ssa") || testCase.formatPattern.equals("hh:mm:ss a") ||
+             testCase.formatPattern.equals("HH:mm:ss") || testCase.formatPattern.equals("hh:mma") ||
+             testCase.formatPattern.equals("hh:mm a") || testCase.formatPattern.equals("HH:mm")){
+            today = LocalDateTime.of(1970, 1, 1, now.getHour(), now.getMinute(), now.getSecond());
+            yesterday = LocalDateTime.of(1970, 1, 1, 0, 0, 0);
+            tomorrow = LocalDateTime.of(1970, 1, 1, 23, 59, 59);
+        } else{
+            today = now;
+            yesterday = today.minusDays(1).withHour(0).withMinute(0).withSecond(0);
+            tomorrow = today.plusDays(1).withHour(23).withMinute(59).withSecond(59);
+        }
+
+
+        Contentlet contentlet = TestDataUtils.getNewsContent(false, languageId, contentType.id());
+        contentlet.setDateProperty("sysPublishDate", Date.from(today.atZone(ZoneId.systemDefault()).toInstant()));
+
+        contentlet = ContentletDataGen.checkin(contentlet, IndexPolicy.WAIT_FOR);
```

Comment:
maybe we can move this code for a private methid llike 'createContent', to have a readable method